### ISSUE TITLE
Get rid of ocaml syntax in dune/jbuild files

### ISSUE THREAD

**Issue by rgrinberg** [ocaml/dune#984]
OCaml syntax in jbuild and dune files can be replaced with a stanza for statically evaluated scripts. For example, we could have something like:

```
(eval-ocaml foo.ml)
```

Currently, there would be limitations to only static paths as arguments (like `include`). But eventually, such limitations might be lifted without breaking user code.

**Comment by Leonidas-from-XIV** [comment]
@rgrinberg As we were going through this and looked at the [documentation](https://dune.readthedocs.io/en/stable/advanced/ocaml-syntax.html) which states

> Consequently you must not build complex systems based on it.

Maybe the way to deal with this is not to move the behavior to a different opt-in syntax but to remove it entirely in a new Dune major release and possibly add support for terser generate/run/include?

**Comment by nojb** [comment]
There is one aspect of OCaml syntax that is surprisingly useful and which may be useful to provide independently: as Dune files in OCaml syntax are evaluated each time one launches a build, it provides a place to do various kinds of checks before going on to the build. For example, at LexiFi the build environment needs various configuration steps to be done before launching the build. Moreover, it is easy to forget to redo these steps when switching branches (which can lead to silent bugs such as linking the wrong version of a library or third-party dependency or having the wrong version of a tool in the PATH). We actually piggy-back on a Dune file in OCaml syntax to check whether all configuration steps have been done at each build and warn the user otherwise.

**Comment by Leonidas-from-XIV** [comment]
I see. I was playing around a bit with something like 

```
(executable
  (name verify))

(rule
  (alias default)
  (deps (universe))
  (action
    (run ./verify.exe)))
```

which could potentially do something similar, however it's not that easy getting the alias to always trigger.

**Comment by nojb** [comment]
> however it's not that easy getting the alias to always trigger.

Exactly, but otherwise indeed it is quite close to what I described.

**Comment by rgrinberg** [comment]
Before removing this feature we need to make sure the alternative is adequate. I don't think we've done that yet.
